components/browser/engine-system/src/main/java/mozilla/components/browser/engine/system/SystemEngineView.kt

```diff
@@ -566,7 +566,8 @@ class SystemEngineView @JvmOverloads constructor(
     internal fun createDownloadListener(): DownloadListener {
         return DownloadListener { url, userAgent, contentDisposition, mimetype, contentLength ->
             session?.internalNotifyObservers {
-                val fileName = DownloadUtils.guessFileName(contentDisposition, url, mimetype)
+
+                val fileName = DownloadUtils.guessFileName(contentDisposition, null, url, mimetype)
```

Comment:
Passing `null` in for `destinationDirectory` here worries me a little bit. This means that in WebView implementations, we wouldn't be able to guess the file name? I'm not sure I understand the significance of `SystemEngineView` nor the purpose of this `DownloadListener`. 

Can you help me understand if this is an issue or if there's a work around I can use here without the `destinationDirectory`?

Reply:
>  I'm not sure I understand the significance of SystemEngineView nor the purpose of this DownloadListener.

Yes, to support downloads in our system-engine (webview).

You're not changing any behaviour here though as you fallback to the same filename that would've been used before. Can we just use `Environment.DIRECTORY_DOWNLOADS` here as default?

Reply:
I'm going to opt for keeping this `null`. I don't want to make it `Environment.DIRECTORY_DOWNLOADS` because that will be another place someone has to change if that default directory ever changes. It'd be weird if these were out of sync :)